src/index.ts

```diff
@@ -13,7 +15,7 @@ const movieSearchBox = document.getElementById(
 const fetchMovieList = async (searchTerm: string): Promise<MovieSearch> => {
   const API_URL = `https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&query=${searchTerm}`
   const movieResp = await fetch(API_URL)
-  return movieResp.json() as Promise<MovieSearch>
+  return (await movieResp.json()) as Promise<MovieSearch>
```

Comment:
So you're thinking it should more like: `await (movieResp.json() as Promise<MovieSearch>)`

Reply:
After looking back through it, this is just a ESLint prettier/prettier issue. The IDE is set to format on save, and that's what keeps happening. I've also tried disabling the linter for this, but it still surrounds the  'await movieResp.json' with (). 

<img width="834" alt="Screen Shot 2023-06-21 at 10 53 24 AM" src="https://github.com/tyseawood/Flick-fil-a/assets/62930969/8e7b6628-8f48-4501-9f19-7be68bbcff2e">

Reply:
Ahhh I see what is happening here. Your original line:
```js
return (await movieResp.json()) as Promise<MovieSearch>
```
is _technically_ fine. 

What's confusing about it is that it's weird, because here's what is happening:
1. `movieResp.json()` returns a `Promise<MovieSearch>`
2. `await movieResp.json()` thus returns a `MovieSearch`
3. but since `fetchMovieList` is `async`, it will return a Promise no matter what, so ultimately `return (await movieResp.json()) as Promise<MovieSearch>` works because it's equivelant to `return MovieSearch as Promise<MovieSearch>`, which is stupid but accurate because `fetchMovieList` is going to make whatever you return into a promise. 

So Promises are kinda dumb haha. It might make more sense to instead just `return movieResp.json() as Promise<MovieSearch>` because then we skip the Promise->Value->Promise back and forth. Ultimately it's the same effect but just thought it was interesting enough to dive into. A good test of JS understanding I guess. 

The parens is not really important here, that's probably just the linter trying to remove some of this ambiguity in order of operations.

Reply:
You can decide if you want to change it. I think it would be a little less confusing but ultimately not a huge deal either way so I'll approve this so you can move on if you want

Reply:
Thanks for breaking that down! I'll leave it be, since when I run the format script it'll do it anyway.